### PR TITLE
Kinetis lpc sdcard

### PR DESCRIPTION
**Pull request by acassis**

### CODE REVIEW

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 1]
```diff
@@ -317,7 +322,8 @@ static inline uint32_t lpc17_40_getpwrctrl(void);
 static void lpc17_40_sampleinit(void);
 static void lpc17_40_sdcard_sample(struct lpc17_40_sdcard_regs_s *regs);
 static void lpc17_40_sample(struct lpc17_40_dev_s *priv, int index);
-static void lpc17_40_sdcard_dump(struct lpc17_40_sdcard_regs_s *regs, const char *msg);
+static void lpc17_40_sdcard_dump(struct lpc17_40_sdcard_regs_s *regs,
+                                 const char *msg);
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:326]: Indentation style is not consistent.  In all other prototypes, the argument that continue on the same coleumn.  Like the following.

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 2]
```diff
@@ -317,7 +322,8 @@ static inline uint32_t lpc17_40_getpwrctrl(void);
...
 static void lpc17_40_dumpsample(struct lpc17_40_dev_s *priv,
               struct lpc17_40_sampleregs_s *regs, const char *msg);
 static void lpc17_40_dumpsamples(struct lpc17_40_dev_s *priv);
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:329]: This is the correct alignment to use in this file.

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 3]
```diff
@@ -305,9 +308,11 @@ struct lpc17_40_sampleregs_s
 static void lpc17_40_takesem(struct lpc17_40_dev_s *priv);
 #define     lpc17_40_givesem(priv) (nxsem_post(&priv->waitsem))
 static inline void lpc17_40_setclock(uint32_t clkcr);
-static void lpc17_40_configwaitints(struct lpc17_40_dev_s *priv, uint32_t waitmask,
+static void lpc17_40_configwaitints(struct lpc17_40_dev_s *priv,
+                                    uint32_t waitmask,
               sdio_eventset_t waitevents, sdio_eventset_t wkupevents);
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:313]: This indentation style is inconstistent with the indentation style used elsewhere (see below)

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 4]
```diff
@@ -334,13 +340,16 @@ static void lpc17_40_dmacallback(DMA_HANDLE handle, void *arg, int status);
...
+static void lpc17_40_dataconfig(uint32_t timeout, uint32_t dlen,
+                                uint32_t dctrl);
 static void lpc17_40_datadisable(void);
 static void lpc17_40_sendfifo(struct lpc17_40_dev_s *priv);
 static void lpc17_40_recvfifo(struct lpc17_40_dev_s *priv);
 static void lpc17_40_eventtimeout(int argc, uint32_t arg);
-static void lpc17_40_endwait(struct lpc17_40_dev_s *priv, sdio_eventset_t wkupevent);
-static void lpc17_40_endtransfer(struct lpc17_40_dev_s *priv, sdio_eventset_t wkupevent);
+static void lpc17_40_endwait(struct lpc17_40_dev_s *priv,
+                             sdio_eventset_t wkupevent);
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:350]: Same comment as above

Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:350]: Same comment as above.

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 5]
```diff
@@ -368,8 +377,9 @@ static int  lpc17_40_attach(FAR struct sdio_dev_s *dev);
 
 static int  lpc17_40_sendcmd(FAR struct sdio_dev_s *dev, uint32_t cmd,
               uint32_t arg);
-static int  lpc17_40_recvsetup(FAR struct sdio_dev_s *dev, FAR uint8_t *buffer,
-              size_t nbytes);
+static int  lpc17_40_recvsetup(FAR struct sdio_dev_s *dev,
+                               FAR uint8_t *buffer,
+                               size_t nbytes);
 static int  lpc17_40_sendsetup(FAR struct sdio_dev_s *dev,
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:383]: Same comment as above.

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c [hunk 6]
```diff
@@ -2290,32 +2338,44 @@ static sdio_eventset_t lpc17_40_eventwait(FAR struct sdio_dev_s *dev,
...
+      if (ret < 0)
+        {
+          wd_cancel(priv->waitwdog);
+
+          leave_critical_section(flags);
+
+          return SDIOWAIT_ERROR;
+        }
+
       wkupevent = priv->wkupevent;
```
Comment (patacongo) [inline arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:2374]: LGTM

File: arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c (hunk 1 above)
Comment (acassis) [reply arch/arm/src/lpc17xx_40xx/lpc17_40_sdcard.c:326]: Hmm, I missed the column start position, normally I use as reference the "(" in the previous line. But some cases like ABIGMACRONAMEA | ABIGMACRONAMEB I try to keep it aligned the beginning of the first macro name. I will fix it

**Pull request merged**